diff -urN fontforge-20120731-b/gutils/gimagereadgif.c fontforge-20120731-b-patched/gutils/gimagereadgif.c
--- fontforge-20120731-b/gutils/gimagereadgif.c	2012-08-03 02:09:09.000000000 +1100
+++ fontforge-20120731-b-patched/gutils/gimagereadgif.c	2014-05-29 21:07:24.771043752 +1100
@@ -218,15 +218,24 @@
 GImage *GImageReadGif(char *filename) {
     GImage *ret, **images;
     GifFileType *gif;
+    int gifErrCode;
     int i;
 
+#if GIFLIB_MAJOR >= 5
+    if ((gif = DGifOpenFileName(filename, &gifErrCode)) == NULL) {
+#else
     if ((gif = DGifOpenFileName(filename)) == NULL) {
+#endif
 	fprintf( stderr, "can't open %s\n", filename);
 return( NULL );
     }
 
     if ( DGifSlurp(gif)==GIF_ERROR ) {
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+	DGifCloseFile(gif, NULL);
+#else
 	DGifCloseFile(gif);
+#endif
 	fprintf(stderr,"Bad gif file %s\n", filename );
 return( NULL );
     }
@@ -238,7 +247,11 @@
 	ret = images[0];
     else
 	ret = GImageCreateAnimation(images,gif->ImageCount);
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+    DGifCloseFile(gif, NULL);
+#else
     DGifCloseFile(gif);
+#endif
     free(images);
 return( ret );
 }
